packages/react-magma-dom/src/components/ToggleButton/ToggleButton.tsx

```diff
@@ -0,0 +1,203 @@
+   */
+  size?: ButtonSize;
+}
+
+export type ToggleButtonProps = XOR<
+  ToggleButtonTextProps,
+  ToggleButtonIconProps
+>;
+
+//Sets the icon width for icon only Toggle Buttons.
+export function setIconWidth(props: ToggleButtonIconProps) {
+  if (props.size === ButtonSize.small) {
+    return props.theme.spaceScale.spacing07;
+  }
+  if (props.size === ButtonSize.large) {
+    return props.theme.spaceScale.spacing11;
+  }
+  return props.theme.spaceScale.spacing09;
+}
+
+//Sets the background colors for the Toggle Buttons.
+export function setBackgroundColor(props) {
+  //Active background color.
+  if (props.selected && props.isChecked) {
+    return 'none';
+  }
+  if (props.selected || props.isChecked) {
+    if (props.isInverse) {
+      return transparentize(0.5, props.theme.colors.neutral900);
+    }
+    return transparentize(0.5, props.theme.colors.neutral300);
+  }
+}
+
+export const ToggleButtonStyles = props => css`
+  background: ${setBackgroundColor(props)};
+  &:not(:disabled):hover {
+    background: ${setBackgroundColor(props)};
+  }
+  &:not(:disabled):focus {
+    background: ${setBackgroundColor(props)};
+    outline-offset: -2px;
+  }
+  &:hover {
+    background: ${setBackgroundColor(props)};
+  }
+`;
+
+const StyledToggleButtonIcon = styled(IconButton)`
+  ${ToggleButtonStyles}
+  min-width: auto;
+  width: ${props => (!props.hasLabel ? setIconWidth : '')};
+  span {
+    padding: ${props => (!props.hasLabel ? 0 : '')};
+  }
+`;
+
+const StyledToggleButtonText = styled(Button)`
+  ${ToggleButtonStyles}
+`;
+
+export const ToggleButton = React.forwardRef<
+  HTMLButtonElement,
+  ToggleButtonProps
+>((props, ref) => {
+  const {
+    'aria-label': ariaLabel,
+    children,
+    disabled,
+    icon,
+    isInverse,
+    onClick,
+    testId,
+    value,
+  } = props;
+
+  const context = React.useContext(ToggleButtonGroupContext);
+
+  const theme = React.useContext(ThemeContext);
+
+  const [selected, setSelected] = React.useState(false);
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    if (!context.exclusive) {
+      if (selected || props.isChecked) {
+        setSelected(false);
+      } else if (!selected) {
+        setSelected(true);
+      }
+    }
+    onClick && typeof onClick === 'function' && onClick(event);
+  };
+
+  return (
+    <>
+      {icon ? (
+        <StyledToggleButtonIcon
+          aria-checked={selected}
+          aria-label={ariaLabel}
+          color={ButtonColor.subtle}
+          disabled={disabled}
+          theme={theme}
+          hasLabel={children ? true : false}
+          icon={icon}
+          id={context.descriptionId}
+          isChecked={
+            context.selectedValue ? context.selectedValue === value : null
+          }
+          isInverse={context.isInverse || isInverse}
+          onClick={handleClick}
+          ref={ref}
+          enforced={context.enforced}
+          role="switch"
```

Comment:
Is the role `switch` even when exclusive is true?

Reply:
Should it be something else you think?

Reply:
I'm not 100% on the best accessibility behavior here, I'd triple check MUI/other similar components. Without researching, in my brain it's a "switch" when you can select multiple and "radio" when it's exclusive.

Reply:
I don't see `role` being used in other component libraries really. I could just make it go between `switch` and `radio` depending on if the prop `exclusive` is used or not.